BranchMetrics/react-native-branch-deep-linking-attribution
# React Native Testbed App

## Reference
SDK-1808 -- [React Native] Develop testbed app

## Summary
(Re)Introduces a testbed app for the branch React Native SDK. 

Contains primary sdk functions:
- Create Branch Link
- Share Branch Link
- Create QR Code
- View install params
- View latest params
- Set user id
- Clear user id
- Disable/Enable tracking
- Send Commerce Event
- Send Content Event
- Send Lifecycle Event
- Send View Event 

100% of the UI is written in Typescript, and only minimal changes to the native implementations to support.

Also fixed  `isTrackingDisabled: () => Promise<boolean>;`, this call needs a response from the native layer which is async when crossing back on the js bridge.

## Motivation
Presently there is no consistent app for internal developers to exercise the SDK's functions. Projects prior fell out of maintenance and were more work to recover than to rewrite.


![image](https://user-images.githubusercontent.com/94577415/225354907-5a77ca27-5bb1-4d8e-a978-9c2f301e1804.png)
![image](https://user-images.githubusercontent.com/94577415/225355006-a06251a0-ff64-4b52-9217-4d030ed95593.png)
![image](https://user-images.githubusercontent.com/94577415/225358028-03eb1e98-df30-45ac-99c4-6c1648644e2b.png)
![image](https://user-images.githubusercontent.com/94577415/225357694-80e9cf9b-0045-4767-9212-697caa461164.png)
![image](https://user-images.githubusercontent.com/94577415/225357775-9e6ef9ec-8f42-4f61-b148-8abdd0971fa6.png)
 

## Type Of Change



## Testing Instructions
Check out the branch, run the test bed app manually and see it performs Branch functions as expected.









cc @BranchMetrics/saas-sdk-devs for visibility.
